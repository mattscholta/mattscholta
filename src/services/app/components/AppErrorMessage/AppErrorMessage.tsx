import * as React from 'react';
import classnames from 'classnames';

interface AppErrorMessageProps {
  className?: string;
  message?: string;
}

/**
 * @name AppErrorMessage
 * @description Used to display errors to our user.
 */
const AppErrorMessage: React.FC<AppErrorMessageProps> = (props) => {
  const { className, message } = props;

  // Styles
  const tailwind = 'u-block u-text-14';
  const cssComponent = classnames(tailwind, className);

  // ðŸ”Œ Short Circuit
  if (!message) return null;

  return <div className={cssComponent}>{message}</div>;
};

export { AppErrorMessage };
export type { AppErrorMessageProps };
