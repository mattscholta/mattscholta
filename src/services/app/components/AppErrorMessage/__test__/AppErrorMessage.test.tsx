import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { AppErrorMessage, AppErrorMessageProps } from '../AppErrorMessage';

describe('AppErrorMessage', () => {
  const message = 'Sample error message';

  let props: AppErrorMessageProps;
  let wrapper: RenderResult;

  const createWrapper = () => <AppErrorMessage {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('returns null when there is no message', () => {
    expect(wrapper.container.firstChild).toBeNull();
    expect(wrapper.container.childNodes).toMatchSnapshot('null');
  });

  it('renders an error message', () => {
    props.message = message;
    wrapper.rerender(createWrapper());

    expect(wrapper.findAllByText(message));
    expect(wrapper.container.childNodes).toMatchSnapshot('rendered');
  });
});
