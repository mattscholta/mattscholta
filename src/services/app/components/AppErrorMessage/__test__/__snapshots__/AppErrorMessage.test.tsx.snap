// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppErrorMessage renders an error message: rendered 1`] = `
NodeList [
  <div
    class="u-block u-text-14"
  >
    Sample error message
  </div>,
]
`;

exports[`AppErrorMessage returns null when there is no message: null 1`] = `NodeList []`;
